### PR TITLE
Super admin gherkins

### PR DESCRIPTION
**Pull request by DXCanas**
Adding gherkins from [here](https://docs.google.com/document/d/1xrRq0UTVn0bum5YuPRQ4ZSNkgpdy5Ny4tzdvgS7lBfg/edit), refactoring them for `super admin` over `superuser`

### CODE REVIEW

**Comment by indirectlylit** [comment]
thank you! @radinamatic would you mind reviewing?

File: integration_testing/features/super-admin/super-admin-downgrade-from-super-admin-to-device-permissions.feature [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Feature: Super admin changes device permissions from super admin to device permissions
```
Comment (radinamatic) [inline integration_testing/features/super-admin/super-admin-downgrade-from-super-admin-to-device-permissions.feature]: Scenario is well described, but the filename and the feature name is a bit unclear. Maybe it's our internal convention at this moment, but "downgrading _to_ device permissions", sounds unclear to me. Technically it's a group/type of permissions, for the moment we have just one but that might change in the future. Could we change this to "content import permission"?

File: integration_testing/features/super-admin/super-admin-downgrade-from-super-admin-to-device-permissions.feature [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+Background:
+  Given I am signed in to Kolibri as a super admin
+    And I am on the *Device > Permissions* page
+    And I have clicked *Edit Permissions* on another super admin
+
+  Scenario: Super admin can see what will happen if they change super admin permissions to device permissions
+    Given I have deselected the *Make super admin* checkbox
+      And I see that all checkboxes are deselected
+      And I see that all checkboxes also become active
```
Comment (radinamatic) [inline integration_testing/features/super-admin/super-admin-downgrade-from-super-admin-to-device-permissions.feature]: Nitpicking, but technically it's just one that becomes active :stuck_out_tongue: 
Not sure if spelling it out in full is necessary (it's long), it's just that the sequence where all checkboxes are deselected, and then they are active sounded weird...

File: integration_testing/features/super-admin/super-admin-downgrade-from-super-admin-to-device-permissions.feature [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+      And I see that the *Save changes* button becomes active
+    When I click on a checkbox under *Device permissions*
+    Then I should see that I can still *Save changes*
```
Comment (radinamatic) [inline integration_testing/features/super-admin/super-admin-downgrade-from-super-admin-to-device-permissions.feature]: Is step necessary? I mean is there a scenario where you would be able to change a state of the checkbox, but not save changes?

File: integration_testing/features/super-admin/super-admin-edit-user.feature [hunk 4]
```diff
@@ -0,0 +1,87 @@
+Feature: Superuser edit users
+    Superuser needs to be able to edit user's full name and username, reset the passwords, change the user types, and delete them from the facility
+
+  Background:
+    Given I am signed in to Kolibri as a super admin
+      And I am on the *Facility > Users* page
+
+  Scenario: Edit user's full name
+    When I click on *Manage* button for the user I want to edit
```
Comment (radinamatic) [inline integration_testing/features/super-admin/super-admin-edit-user.feature]: Is this on latest develop? It should be `Options` button and `Edit details` option in the step below...

ledev18aut running - oracle vm virtualbox_220

You should review the rest of the scenario too.

File: integration_testing/features/super-admin/super-admin-grant-super-admin-permissions.feature [hunk 5]
```diff
@@ -0,0 +1,21 @@
+Feature: Super admin gives super admin permissions to a facility user
+  A super admin should understand the implications of what it means to
+make a facility user a super admin
```
Comment (radinamatic) [inline integration_testing/features/super-admin/super-admin-grant-super-admin-permissions.feature:3]: Who exactly is **facility** user, any user of the facility or a **Facility admin**?

File: integration_testing/features/super-admin/super-admin-revoke-super-admin-permissions.feature [hunk 6]
```diff
@@ -0,0 +1,18 @@
+Feature: Super admin removes all device permissions from another super admin
+
+  Background:
+    Given I am signed in to Kolibri as a super admin
+      And I am on the *Device > Permissions* page
+      And I have clicked *Edit Permissions* on another super admin
+
+  Scenario: Super admin can see what will happen if they want to remove super admin permissions
+    When I deselect the *Make super admin checkbox*
```
Comment (radinamatic) [inline integration_testing/features/super-admin/super-admin-revoke-super-admin-permissions.feature]: `When I deselect the *Make super admin* checkbox`

File: integration_testing/features/super-admin/super-admin-revoke-super-admin-permissions.feature [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    Then I see that checkboxes under *Device permissions* also become unchecked
+      And I see that those checkboxes under *Device permissions also become active
```
Comment (radinamatic) [inline integration_testing/features/super-admin/super-admin-revoke-super-admin-permissions.feature]: Missing `*` after "permissions".

**Review by radinamatic — CHANGES_REQUESTED** [review]
Welcome @DXCanas to the pickling club! :tada: 

Looking good, just a few comments :+1:

File: integration_testing/features/super-admin/super-admin-downgrade-from-super-admin-to-device-permissions.feature (hunk 1 above)
Comment (DXCanas) [reply integration_testing/features/super-admin/super-admin-downgrade-from-super-admin-to-device-permissions.feature]: Is "downgrade" is still acceptable?

Comment (DXCanas) [reply integration_testing/features/super-admin/super-admin-downgrade-from-super-admin-to-device-permissions.feature]: Also, what do you think of "limited device permissions"? To maintain consistency between the UI and the gherkin?

File: integration_testing/features/super-admin/super-admin-downgrade-from-super-admin-to-device-permissions.feature (hunk 3 above)
Comment (DXCanas) [reply integration_testing/features/super-admin/super-admin-downgrade-from-super-admin-to-device-permissions.feature]: @jtamiace

File: integration_testing/features/super-admin/super-admin-edit-user.feature (hunk 4 above)
Comment (DXCanas) [reply integration_testing/features/super-admin/super-admin-edit-user.feature]: Didn't update every gherkin story; by and large left the existing ones alone. I did rename files, however, which is why they're showing up here as "new". I'll update this, though.

File: integration_testing/features/super-admin/super-admin-grant-super-admin-permissions.feature (hunk 5 above)
Comment (DXCanas) [reply integration_testing/features/super-admin/super-admin-grant-super-admin-permissions.feature:3]: Any user on the facility. This is carryover from the time before `superuser` - when device owners where facility agnostic.

@jtamiace thoughts?

**Review by DXCanas — COMMENTED** [review]
Pinging @jtamiace for contributions

@indirectlylit I'm noticing that it's pretty easy for Gherkins to get out of date (and less useful) with a simple change in copy. The ones I pasted here were written before some string updates happened in the UI, becoming significantly less useful since the strings act as identifiers for an integration tester. Don't really have any solutions, just flagging a shortcoming... thoughts?

File: integration_testing/features/super-admin/super-admin-grant-super-admin-permissions.feature (hunk 5 above)
Comment (jtamiace) [reply integration_testing/features/super-admin/super-admin-grant-super-admin-permissions.feature:3]: Yep, any user of the facility (learner, coach, or facility admin)

**Comment by indirectlylit** [comment]
>  I'm noticing that it's pretty easy for Gherkins to get out of date (and less useful) with a simple change in copy

that's a great point.

I guess the two options are either

1. find a way to abstract them from actual text. Not sure if that's possible... (@radinamatic)
2. make it a matter of process to search stories when updating user-facing text

I think 2 is probably more likely what we'll need to do...

File: integration_testing/features/super-admin/super-admin-downgrade-from-super-admin-to-device-permissions.feature (hunk 1 above)
Comment (radinamatic) [reply integration_testing/features/super-admin/super-admin-downgrade-from-super-admin-to-device-permissions.feature]: "Downgrade" sounds Ok to me (anything below the super admin is _down_ :wink:), as does  "limited device permissions",  but "content import permission" is more explicit, at least at this stage.

File: integration_testing/features/super-admin/super-admin-edit-user.feature (hunk 4 above)
Comment (radinamatic) [reply integration_testing/features/super-admin/super-admin-edit-user.feature]: Understood, I think that is a fair choice, write the new ones according to the current state of things, and update the rest in a separate PR (not you necessarily).

File: integration_testing/features/super-admin/super-admin-grant-super-admin-permissions.feature (hunk 5 above)
Comment (radinamatic) [reply integration_testing/features/super-admin/super-admin-grant-super-admin-permissions.feature:3]: Gotcha!

**Comment by radinamatic** [comment]
> 1. find a way to abstract them from actual text. Not sure if that's possible...

Not that I've seen, but will research more!

> 2. make it a matter of process to search stories when updating user-facing text

Maybe add a `gherkin-review` label on each PR that involves a change in the copy, so that they can be filtered once merged in order to update the scenarios?

**Review by radinamatic — APPROVED** [review]
Looking good! :+1:

**Comment by indirectlylit** [comment]
thanks all!

**Pull request merged**